compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -132,7 +132,25 @@ fn get_llvm_object_symbols(
     if err.is_null() {
         return Ok(true);
     } else {
-        return Err(unsafe { *Box::from_raw(err as *mut io::Error) });
+        let error = unsafe { *Box::from_raw(err as *mut String) };
+        // These are the magic constants for LLVM bitcode files:
+        // https://github.com/llvm/llvm-project/blob/7eadc1960d199676f04add402bb0aa6f65b7b234/llvm/lib/BinaryFormat/Magic.cpp#L90-L97
+        if buf.starts_with(&[0xDE, 0xCE, 0x17, 0x0B]) || buf.starts_with(b"BC\xC0\xDE") {
```

Comment:
why the two different ways of writing a byte string?

Reply:
I could write the first one as `b""` too, but I did have to use `\x` escapes for all bytes in that case.

Reply:
or you could write the second one as an array

Reply:
Done